### PR TITLE
Fixes #8981: Cannot save new generic method with some contraints

### PR DESCRIPTION
**Pull request by RaphaelGauthier**
https://www.rudder-project.org/redmine/issues/8981

### CODE REVIEW

File: builder/js/ncf.js [hunk 1]
```diff
@@ -89,6 +89,9 @@ app.directive('constraint', function($http, $q, $timeout) {
     },
     require: 'ngModel',
     link: function(scope, elm, attrs, ctrl) {
+    if(scope.parameter.constraints.allow_empty_string){
```
Comment (VinceMacBuche) [inline builder/js/ncf.js]: You need to check if the value is undefined before unless value will always be empty

**Comment by RaphaelGauthier** [comment]
PR updated

File: builder/js/ncf.js (hunk 1 above)
Comment (RaphaelGauthier) [inline builder/js/ncf.js]: Done!

File: builder/js/ncf.js [hunk 2]
```diff
@@ -89,6 +89,9 @@ app.directive('constraint', function($http, $q, $timeout) {
     },
     require: 'ngModel',
     link: function(scope, elm, attrs, ctrl) {
+    if((scope.parameter.constraints.allow_empty_string)&&(scope.parameter.value==null)){
```
Comment (VinceMacBuche) [inline builder/js/ncf.js]: should check for null but ( scope.parameter.value === undefined)

And ndever double equal ;)

Comment (RaphaelGauthier) [inline builder/js/ncf.js]: I thought it was better to check both cases, null and undefined. But ok, it's done!

**Comment by RaphaelGauthier** [comment]
PR updated

**Pull request merged**